Add new flag --remote_default_platform_properties.

katre opened the pull request:
This will replace --host_platform_remote_properties_override.

Fixes #7166.

katre commented:
Still not sure what's causing the errors, will debug more later.

buchgr commented on src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java:
```diff
@@ -413,18 +415,31 @@ static Action buildAction(
+      try {
+        TextFormat.getParser().merge(executionPlatform.remoteExecutionProperties(), platformBuilder);
+      } catch (ParseException e) {
+        throw new IllegalArgumentException(
```
I think it would be better to not use a runtime exception, as this will kill the Bazel server. Prefer an `AbruptExitException` or similar to provide a user friendly error.

katre replied:
Change to UserExecError, since the various remote classes already throw and handle ExecError.

katre commented:
Fixed the error. It turns out that if the Platform is always present in the API call, but empty, some tests break. I assume this is due to test setup and not the actual API semantics, but I have fixed it for now.

buchgr commented:
LGTM
